packages/button/src/react/index.tsx

```diff
@@ -0,0 +1,237 @@
+        size: mapIconSize(props.size)
+      })}
+    </div>
+  ) : null
+
+renderIcon.propTypes = {
+  loading: PropTypes.bool,
+  icon: PropTypes.element
+}
+
+interface ButtonStatics {
+  appearances: typeof vars.appearances
+  iconAligns: typeof vars.iconAligns
+  sizes: typeof vars.sizes
+}
+
+interface ButtonProps extends HTMLAttributes<HTMLElement> {
+  appearance?: string
+  disabled?: boolean
+  href?: string
+  icon?: React.ReactNode
+  iconAlign?: string
+  loading?: boolean
+  size?: string
+}
+
+interface ButtonComponent
+  extends RefForwardingComponent<ButtonProps, HTMLElement, ButtonStatics> {}
+
+const Button = React.forwardRef<HTMLElement, ButtonProps>(
+  (props, forwardedRef) => {
+    const themeName = useTheme()
+    const ref = React.useRef<HTMLAnchorElement | HTMLButtonElement>()
+    React.useImperativeHandle(forwardedRef, () => ref.current)
+    const nonLoadingWidth = React.useMemo(() => {
+      if (props.loading && ref && ref.current) {
+        return ref.current.offsetWidth
+      }
+    }, [props.loading, ref])
+
+    const tagName = props.href ? 'a' : 'button'
+    const isLoadingWithNoText = !!nonLoadingWidth
+    const allProps = {
+      ...props,
+      isLoadingWithNoText,
+      iconOnly: React.Children.count(props.children) <= 0,
+      themeName
+    }
+
+    const isDisabledLink = allProps.disabled && allProps.href
+    const filteredProps = filterReactProps(props, { tagName })
+    delete filteredProps.icon
+    if (isDisabledLink) {
+      delete filteredProps.onClick
+    }
+
+    return React.createElement(
+      tagName,
+      {
+        ...styles.button({
+          appearance: allProps.appearance,
+          disabled: allProps.disabled,
+          icon: allProps.icon,
+          iconAlign: allProps.iconAlign,
+          iconOnly: allProps.iconOnly,
+          size: allProps.size,
+          themeName: allProps.themeName
+        }),
+        ...filteredProps,
+        disabled: props.disabled || props.loading,
+        ref,
+        style: isLoadingWithNoText
+          ? { ...props.style, width: nonLoadingWidth }
+          : props.style || {}
+      },
+      renderIcon({
+        loading: allProps.loading,
+        icon: allProps.icon,
+        appearance: allProps.appearance,
+        themeName: allProps.themeName,
+        size: allProps.size,
+        iconOnly: allProps.iconOnly,
+        isLoadingWithNoText: allProps.isLoadingWithNoText,
+        iconAlign: allProps.iconAlign
+      }),
+      !isLoadingWithNoText && (
+        <span {...styles.text()}>{allProps.children}</span>
+      )
+    )
+  }
+) as ButtonComponent
+
+Button.propTypes = {
+  appearance: PropTypes.oneOf(Object.keys(vars.appearances)),
+  children: PropTypes.any,
+  disabled: PropTypes.bool,
+  href: PropTypes.string,
+  icon: PropTypes.element,
+  iconAlign: PropTypes.oneOf(Object.keys(vars.iconAligns)),
+  loading: PropTypes.bool,
+  size: PropTypes.oneOf(Object.keys(vars.sizes)),
+  style: PropTypes.object
+}
```

Comment:
```suggestion
Button.propTypes = {
  appearance: PropTypes.oneOf(Object.keys(vars.appearances)),
  children: PropTypes.any,
  disabled: PropTypes.bool,
  href: PropTypes.string,
  icon: PropTypes.element,
  iconAlign: PropTypes.oneOf(Object.keys(vars.iconAligns)),
  loading: PropTypes.bool,
  size: PropTypes.oneOf(Object.keys(vars.sizes)),
  style: PropTypes.object
}
```

Reply:
i don't think we need to add proptypes anymore. if we want to support this in the future we should generate them from the ts typings